odc/stats/tasks.py

```diff
@@ -151,18 +158,65 @@ def match_dss(groups, group_size):
         )
         grouped_dss = match_dss(grouped_dss, group_size)
 
-        def map_fuse_func(x):
-            return fuse_ds(*x, product=product)
+        if fuse_dss:
+            fused_product = fuse_products(*products)
+
+            def map_fuse_func(x):
+                return fuse_ds(*x, product=fused_product)
+
+            dss = map(map_fuse_func, grouped_dss)
+            return dss
+        else:
+            return grouped_dss
+
+    @classmethod
+    def _find_dss(
```

Comment:
Having `_find_dss` and `_get_dss` side by side like this is potentially confusing.